src/controller.js

```diff
@@ -126,66 +121,12 @@ DebugletApi.prototype.registerError = function(message) {
  * @param {!function(?Error,Object=)} callback
  * @private
  */
-DebugletApi.prototype.register_ = function(errorMessage, callback) {
+Controller.prototype.register_ = function(errorMessage, callback) {
   var that = this;
-
-  var cwd = process.cwd();
-  var mainScript = path.relative(cwd, process.argv[1]);
-
-  var version = 'google.com/node-' +
-    (that.onGCP ? 'gcp' : 'standalone') +
-    '/v' + pjson.version;
-  var desc = process.title + ' ' + mainScript;
-  var labels = {
-    'main script': mainScript,
-    'process.title': process.title,
-    'node version': process.versions.node,
-    'V8 version': process.versions.v8,
-    'agent.name': pjson.name,
-    'agent.version': pjson.version,
-    'projectid': that.project_
-  };
-
-  var serviceName = that.serviceName_;
-  if (serviceName) {
-    desc += ' module:' + serviceName;
-    labels[DEBUGGEE_MODULE_LABEL] = serviceName;
-  }
-
-  var serviceVersion = that.serviceVersion_;
-  if (serviceVersion) {
-    desc += ' version:' + serviceVersion;
-    if (serviceVersion !== 'default') {
-      labels[DEBUGGEE_MAJOR_VERSION_LABEL] = serviceVersion;
-    }
-  }
-
-  var descriptor = that.descriptor_;
-  if (descriptor) {
-    desc += ' description:' + descriptor;
-  }
-
-  if (process.env.GAE_MINOR_VERSION) {
-    labels[DEBUGGEE_MINOR_VERSION_LABEL] = process.env.GAE_MINOR_VERSION;
-  }
-
-  var uniquifier = desc + version + that.uid_ + that.sourceContext_ +
-    JSON.stringify(labels);
-  uniquifier  = crypto.createHash('sha1').update(uniquifier).digest('hex');
-
-  var debuggee = {
-    project: that.project_,
-    uniquifier: uniquifier,
-    description: desc,
-    agentVersion: version,
-    labels: labels,
-    sourceContexts: [that.sourceContext_]
-  };
-
-  if (errorMessage) {
-    debuggee.status = new StatusMessage(StatusMessage.UNSPECIFIED, errorMessage,
-                                        true);
-  }
+  var debuggee = new Debuggee(
+      that.project_, that.uid_,
```

Comment:
Determinate variable declarations not at the top of the file? Although I would advocate for the consistent use of this in the same scope I'm not sure I agree with moving it out of the top-level of the function